components/raftstore/src/store/worker/life.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::{Engines, KvEngine, RaftEngine};
+
+/// An assistant to help manage the lifecycle of engines.
+///
+/// Currently, it only enables manual compaction jobs before starting the
+/// engines and disables manual compaction jobs before shutting down the
+/// engines.
+pub struct Assistant<E, R>
+where
+    E: KvEngine,
+    R: RaftEngine,
+{
+    _name: String,
+    engines: Engines<E, R>,
+}
+
+impl<E, R> Assistant<E, R>
+where
+    E: KvEngine,
+    R: RaftEngine,
+{
+    pub fn new<S: Into<String>>(name: S, engines: Engines<E, R>) -> Self {
+        Self {
+            _name: name.into(),
+            engines,
+        }
+    }
+
+    pub fn start(&mut self) {
+        // Enable manual compaction jobs before starting the engines.
+        let _ = self.engines.kv.enable_manual_compaction();
+    }
+
+    pub fn stop(&mut self) {
+        // Disable manul compaction jobs before shutting down the engines. And it
+        // will stop the compaction thread in advance, so it won't block the
+        // cleanup thread when exiting.
+        let _ = self.engines.kv.disable_manual_compaction();
```

Comment:
why not just call it when stopping all workers

Reply:
Yep, its the simplest way to implement it. But as comments for `Assistant`, it's built for later developments and enhancements.

IMO, using `Assistant` also simple as expected.

Reply:
But I can't come up with a requirement that needs it in near future